apps/api-extractor/src/generators/ApiModelGenerator.ts

```diff
@@ -669,6 +671,15 @@ export class ApiModelGenerator {
     this._processChildDeclarations(astDeclaration, exportedName, apiInterface);
   }
 
+  private _determineReadonly(astDeclaration: AstDeclaration): boolean {
+    const apiItemMetadata: ApiItemMetadata = this._collector.fetchApiItemMetadata(astDeclaration);
+    const docComment: tsdoc.DocComment | undefined = apiItemMetadata.tsdocComment;
+    const declarationMetadata: DeclarationMetadata = this._collector.fetchDeclarationMetadata(astDeclaration);
+    return (astDeclaration.modifierFlags & ts.ModifierFlags.Readonly) !== 0
```

Comment:
I think this logic is complicated enough that it might be worth splitting up into three separate checks, with each check getting a nice doc comment above it explaining the case it's checking.

Reply:
Comments have been added. Let me know if more verbosity is desired.